### PR TITLE
Support Wan2.2 t2v diffusers quantization

### PR DESCRIPTION
**Pull request by shengliangxu**
## What does this PR do?

**Type of change:**

new feature

**Overview:**

Support Wan2.2 t2v diffusers quantization
1. fix torch2.9 support

2. add Wan2.2 t2v diffusers pipeline quantization

   Main difference of the Wan2.2 pipeline comparing to exisiting
   pipelines is that there are 2 backbone models for denoising. For the
   quantization therefore we need to quantize both of them.

   However, it turns out our base library does not well support
   quantization of multiple models in the same time. Therefore, the
   change here just stick to quantize a single model each time, and then
   run the quantization multiple times.

   So, we need to allow users to pick which backbone to quantize,
   therefore adding a new argment for it

3. add a workaround for the exporting ONNX issue when we upgrade
   diffusers to >= 0.35.0. The issue lies is the exporting of the
   torch.nn.RMSNorm. Some pipelines in the diffusers > 0.35.0 use the
   torch version RMSNorm while before that they use the diffusers' own
   version of RMSNorm. It turns out they are directly replacable so the
   workaround is to simply replace the torch RMSNorm usages with
   diffusers RMSNorm. But we need to fix it properly soon by porting our
   ONNX export to be based on torch dynamo instead of torchscript. Issue
   reported from external user: https://github.com/NVIDIA/TensorRT-Model-Optimizer/issues/262

4. allow use of a prompts file, which is simply a text file with a list
   of prompts, one prompt each line

5. allow each component of a pipeline to have different dtype accuracy.
   added a new list stype command line arg --component-dtype for this.
   example: --component-dtype vae:Float

6. print the summary of the quantized model so users can capture issues from
   log

## Usage

python quantize.py \
    --model wan2.2-t2v-14b \
    --format fp8 \
    --batch-size 4 \
    --calib-size 64 \
    --n-steps 20 \
    --backbone transformer \
    --model-dtype BFloat16 \
    --component-dtype vae:Float \
    --trt-high-precision-dtype BFloat16 \
    --quantized-torch-ckpt-save-path ./wan_transformer.pt \
    --onnx-dir wan-transformer-onnx \
    --prompts-file wan-prompts.txt

## Testing

Tested SDXL_BASE, LTX_VIDEO_DEV, WAN22_T2V

## Before your PR is "*Ready for review*"

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Yes/No 
- **Did you write any new necessary tests?**: Yes/No
- **Did you add or update any necessary documentation?**: Yes/No
- **Did you update [Changelog](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CHANGELOG.rst)?**: Yes/No 

## Additional Information

https://github.com/NVIDIA/TensorRT-Model-Optimizer/issues/262

### CODE REVIEW

File: examples/diffusers/quantization/onnx_utils/export.py [hunk 1]
```diff
@@ -293,7 +302,51 @@ def _gen_dummy_inp_and_dyn_shapes_ltx(backbone, min_bs=2, opt_bs=2):
...
+
+def _gen_dummy_inp_and_dyn_shapes_wan(backbone, min_bs=1, opt_bs=2):
+    assert isinstance(backbone, WanTransformer3DModel)
+    dtype = backbone.dtype
+
+    channels = 16  # latent channels from VAE
+    hidden_size = 4096  # text encoder hidden size (UMT5-XXL)
+
+    min_num_frames = 4 * 1 + 1
+    opt_num_frames = 4 * 20 + 1
```
Comment (ajrasane) [inline examples/diffusers/quantization/onnx_utils/export.py:320]: Could you provide more details about why we write this as 4 * x + 1?

File: examples/diffusers/quantization/onnx_utils/export.py [hunk 2]
```diff
@@ -449,22 +504,34 @@ def modelopt_export_sd(backbone, onnx_dir, model_name, precision):
...
     else:
         raise NotImplementedError(f"Unsupported model_id: {model_name}")
 
     do_constant_folding = True
     opset_version = 20
 
     with quantizer_context, torch.inference_mode():
+        # program = export(backbone, (), dummy_kwargs, dynamic_shapes=dynamic_shapes, strict=False)
+        # print(f"Exported program: {program}")
+        #    dynamic_shapes=dynamic_shapes,
```
Comment (ajrasane) [inline examples/diffusers/quantization/onnx_utils/export.py]: nit: Remove

File: examples/diffusers/quantization/quantize.py [hunk 3]
```diff
@@ -47,12 +60,15 @@
...
     filter_func_default,
     filter_func_ltx_video,
+    filter_func_wan_video,
     load_calib_prompts,
 )
 
 import modelopt.torch.opt as mto
 import modelopt.torch.quantization as mtq
 
+TorchRMSNorm = torch.nn.modules.normalization.RMSNorm
```
Comment (ajrasane) [inline examples/diffusers/quantization/quantize.py]: Do we need this variable?

File: examples/diffusers/quantization/quantize.py [hunk 4]
```diff
@@ -132,27 +161,110 @@ def get_model_filter_func(model_type: ModelType) -> Callable[[str], bool]:
...
 
 # Model-specific default arguments for calibration
 MODEL_DEFAULTS: dict[ModelType, dict[str, Any]] = {
+    ModelType.SDXL_BASE: {
+        "backbone": "unet",
+        "dataset": {
+            "name": "Gustavosta/Stable-Diffusion-Prompts",
+            "split": "train",
+            "column": "Prompt",
+        },
```
Comment (ajrasane) [inline examples/diffusers/quantization/quantize.py:184]: Could you simplify this by creating a helper like:
```python
# Helper to merge dictionaries
def cfg(dataset=True, **kwargs):
    if dataset:
        return {"dataset": DEFAULT_DATASET, **kwargs}
    return kwargs
```

File: examples/diffusers/quantization/onnx_utils/export.py [hunk 5]
```diff
@@ -449,22 +504,34 @@ def modelopt_export_sd(backbone, onnx_dir, model_name, precision):
...
         onnx_export(
```
Comment (ajrasane) [inline examples/diffusers/quantization/onnx_utils/export.py:527]: Are the exported ONNX models runnable with TRT?

File: examples/diffusers/quantization/onnx_utils/export.py (hunk 1 above)
Comment (shengliangxu) [reply examples/diffusers/quantization/onnx_utils/export.py:320]: will do

File: examples/diffusers/quantization/quantize.py (hunk 3 above)
Comment (shengliangxu) [reply examples/diffusers/quantization/quantize.py]: removed.

File: examples/diffusers/quantization/onnx_utils/export.py (hunk 2 above)
Comment (shengliangxu) [reply examples/diffusers/quantization/onnx_utils/export.py]: removed

File: examples/diffusers/quantization/quantize.py (hunk 4 above)
Comment (shengliangxu) [reply examples/diffusers/quantization/quantize.py:184]: I actually feel just have all these config values listed explicitly is clearer. But in the long run, we definitely need to have better config instead of hard code them. How do you think?

File: examples/diffusers/quantization/onnx_utils/export.py (hunk 1 above)
Comment (shengliangxu) [reply examples/diffusers/quantization/onnx_utils/export.py:320]: added sources.

File: examples/diffusers/quantization/onnx_utils/export.py (hunk 5 above)
Comment (shengliangxu) [reply examples/diffusers/quantization/onnx_utils/export.py:527]: I have not actually tried it

**Comment by shengliangxu** [comment]
> Could you create and E2E integration test for this under: https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/tests/examples/diffusers/test_diffusers.py Do we have a smaller variant of the model that we could use for this?

Unfortunately I do not see small models present in HF. These models are really new, so it is normal. We can add testing once small models come out.

File: examples/diffusers/quantization/onnx_utils/export.py (hunk 5 above)
Comment (shengliangxu) [reply examples/diffusers/quantization/onnx_utils/export.py:527]: Just tried trtexec, it hits issues, which is not surprising, the mlperf team is also working on this, I think they need to modify the onnx file using onnx_graphsurgeon. I would leave this part to them. We can then update the codbase to include their solution later. How do you think?
